seatunnel-connectors-v2/connector-hbase/src/main/java/org/apache/seatunnel/connectors/seatunnel/hbase/sink/HbaseSinkWriter.java

```diff
@@ -0,0 +1,189 @@
+import org.apache.seatunnel.api.table.type.SeaTunnelRowType;
+import org.apache.seatunnel.common.exception.CommonErrorCode;
+import org.apache.seatunnel.connectors.seatunnel.common.sink.AbstractSinkWriter;
+import org.apache.seatunnel.connectors.seatunnel.hbase.config.HbaseParameters;
+import org.apache.seatunnel.connectors.seatunnel.hbase.exception.HbaseConnectorException;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hbase.HBaseConfiguration;
+import org.apache.hadoop.hbase.HConstants;
+import org.apache.hadoop.hbase.TableName;
+import org.apache.hadoop.hbase.client.Admin;
+import org.apache.hadoop.hbase.client.BufferedMutator;
+import org.apache.hadoop.hbase.client.BufferedMutatorParams;
+import org.apache.hadoop.hbase.client.Connection;
+import org.apache.hadoop.hbase.client.ConnectionFactory;
+import org.apache.hadoop.hbase.client.Durability;
+import org.apache.hadoop.hbase.client.HTable;
+import org.apache.hadoop.hbase.client.Put;
+import org.apache.hadoop.hbase.util.Bytes;
+
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class HbaseSinkWriter extends AbstractSinkWriter<SeaTunnelRow, Void> {
+
+    private static final String ALL_COLUMNS = "all_columns";
+
+    private final Configuration hbaseConfiguration  = HBaseConfiguration.create();
+
+    private final Connection hbaseConnection;
+
+    private final BufferedMutator hbaseMutator;
+
+    private final Admin hbaseAdmin;
+
+    private final SeaTunnelRowType seaTunnelRowType;
+
+    private final HbaseParameters hbaseParameters;
+
+    private final List<Integer> rowkeyColumnIndexes;
+
+    private final int versionColumnIndex;
+
+    private String defaultFamilyName = "value";
+
+    public HbaseSinkWriter(SeaTunnelRowType seaTunnelRowType,
+                           HbaseParameters hbaseParameters,
+                           List<Integer> rowkeyColumnIndexes,
+                           int versionColumnIndex) throws IOException {
+        this.seaTunnelRowType = seaTunnelRowType;
+        this.hbaseParameters = hbaseParameters;
+        this.rowkeyColumnIndexes = rowkeyColumnIndexes;
+        this.versionColumnIndex = versionColumnIndex;
+
+        if (hbaseParameters.getFamilyNames().size() == 1) {
+            defaultFamilyName = hbaseParameters.getFamilyNames().getOrDefault(ALL_COLUMNS, "value");
+        }
+
+        // initialize hbase configuration
+        hbaseConfiguration.set("hbase.zookeeper.quorum", hbaseParameters.getZookeeperQuorum());
+        if (hbaseParameters.getHbaseExtraConfig() != null) {
+            hbaseParameters.getHbaseExtraConfig().forEach(hbaseConfiguration::set);
+        }
+        // initialize hbase connection
+        hbaseConnection = ConnectionFactory.createConnection(hbaseConfiguration);
+        // initialize hbase admin
+        hbaseAdmin = hbaseConnection.getAdmin();
```

Comment:
Is there any need to create `hbaseAdmin`? I didn't see any use of this field

Reply:
Thank you for your review. This parameter will be used to support save mode feature of sink connectors. But this feature is in the process of development. You can refer to https://github.com/apache/incubator-seatunnel/pull/4022 get more details. If you think at this stage we need remove useless parameters, I will delete it.